go/src/socialapi/workers/payment/subscription.go

```diff
@@ -119,30 +116,45 @@ func EnsureSubscriptionForGroup(groupName string, params *stripe.SubParams) (*st
 		return nil, ErrCustomerNotExists
 	}
 
+	hasSource, err := checkCustomerHasSource(group.Payment.Customer.ID)
+	if err != nil {
+		return nil, err
+	}
+
+	if !hasSource {
+		return nil, ErrCustomerSourceNotExists
```

Comment:
i will use the difference between "having no card" & "not being able to check it" today in another feature